lib/charms/ovn_charm.py

```diff
@@ -1432,3 +1434,41 @@ def assess_exporter(self):
         else:
             snap.install('prometheus-ovs-exporter', channel=channel)
         snap.connect_all()
+
+    def restart_exporter(self, only_if_dead=False):
+        needs_restart = True
+        if only_if_dead:
+            try:
+                service_output = self.run("snap", "services",
```

Comment:
Any reason to not use the already existing `service_running` [0] and `service_restart` [1] functions for this?

0: https://github.com/juju/charm-helpers/blob/81697653d4e4125ebe7b0495041cc70e989765af/charmhelpers/core/host.py#L307-L318
1: https://github.com/juju/charm-helpers/blob/81697653d4e4125ebe7b0495041cc70e989765af/charmhelpers/core/host.py#L144-L168

Reply:
I was looking for this functionality in the `snap` layer, didn't htink to check charm_helpers, thanks for pointing these out.